packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -191,6 +191,97 @@ export async function uploadFile(node: ZoweDatasetNode, doc: vscode.TextDocument
+
+    // Filter out sessions, favorite nodes, or information messages
+    const nodes: IZoweDatasetTreeNode[] = selectedNodes.filter(
+        (selectedNode) =>
+            selectedNode.getParent() &&
+            !contextually.isFavorite(selectedNode) &&
+            !contextually.isFavorite(selectedNode.getParent()) &&
+            !contextually.isSession(selectedNode) &&
+            !contextually.isInformation(selectedNode)
+    );
+
+    // The names of the nodes that should be deleted
+    const deletedNodes: string[] = nodes.map((deletedNode) => {
+        return contextually.isDsMember(deletedNode)
+            ? ` ${deletedNode.getParent().getLabel()}(${deletedNode.getLabel()})`
+            : ` ${deletedNode.getLabel()}`;
+    });
+
+    // Confirm that the user really wants to delete
+    globals.LOG.debug(localize("deleteDatasetPrompt.log.debug", "Deleting data set(s): ") + deletedNodes.join(","));
+    const quickPickOptions: vscode.QuickPickOptions = {
+        placeHolder: localize(
+            "deleteDatasetPrompt.quickPickOption",
+            "Delete data set(s)? This will permanently remove them from your system."
+        ),
+        ignoreFocusOut: true,
+        canPickMany: false,
+    };
+    if (
+        (await vscode.window.showQuickPick(
+            [
+                localize("deleteDatasetPrompt.showQuickPick.delete", "Delete"),
+                localize("deleteDatasetPrompt.showQuickPick.Cancel", "Cancel"),
+            ],
+            quickPickOptions
+        )) !== localize("deleteDatasetPrompt.showQuickPick.delete", "Delete")
+    ) {
+        globals.LOG.debug(
+            localize("deleteDatasetPrompt.showQuickPick.log.debug", "Dataset deletion action was canceled.")
+        );
+        return;
+    }
+
+    if (nodes.length > 0) {
+        // Delete multiple selected nodes
+        await vscode.window.withProgress(
+            {
+                location: vscode.ProgressLocation.Notification,
+                title: localize("deleteDatasetPrompt.deleteCounter", "Deleting nodes"),
+                cancellable: true,
+            },
+            async (progress, token) => {
+                token.onCancellationRequested(() => {
+                    // will be returned as undefined
+                    vscode.window.showInformationMessage(
+                        localize("deleteDatasetPrompt.deleteCancelled", "Delete action was cancelled.")
+                    );
+                });
+                const total = 100;
+                for (const [index, currNode] of nodes.entries()) {
+                    progress.report({
+                        message: `Deleting ${index + 1} of ${nodes.length}`,
+                        increment: (index / nodes.length) * total,
+                    });
+                    try {
+                        await deleteDataset(currNode, datasetProvider);
+                    } catch (err) {
+                        // Do nothing
```

Comment:
Just a suggestion: maybe we can return this error?

So, there will be no super long list of info messages for the user, but one report with success and error items inside.

We can transform this: `await vscode.window.withProgress...` into `const results = await vscode.window.withProgress...` and iterate via the results to show success deletions and error deletions in one message. 

Hope it helps

Reply:
All of the succesful items are already shown in one message, and each failed item has its own message with the error. I somewhat like having each error pop up separately...

Reply:
Oh, this is a good point. I added a fix, see if it's ok please...